airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessBuilderFactory.java

```
@@ -53,6 +54,11 @@ public DockerProcessBuilderFactory(Path workspaceRoot, String workspaceMountSour
 
   @Override
   public ProcessBuilder create(final Path jobRoot, final String imageName, final String... args) {
+
+    if (!checkImageExists(imageName)) {
+      throw new IllegalStateException("Could not find image: " + imageName);
```

You can throw a worker exception here if you want (and chang the iface)